base/compiler/ssair/passes.jl

```diff
@@ -1230,8 +1230,7 @@ function sroa_pass!(ir::IRCode, inlining::Union{Nothing,InliningState}=nothing)
                 # by `stmt.args`. Which have :scope set. In practice, the frontend
                 # does emit these in order, so we could simply go to the last one,
                 # but we want to avoid making that semantic assumption.
-                for i = 1:length(stmt.args)
-                    scope = stmt.args[i]
+                for scope in stmt.args
```

Comment:
This can force allocation of a different Tuple type for every type of object found in the AST, for the iterate protocol, especially when the compiler is not working yet, so it isn't necessarily desirable to combine these